src/aiokatcp/sensor.py

```diff
@@ -187,6 +188,37 @@ def __init__(
         self.auto_strategy_parameters = tuple(auto_strategy_parameters)
         # TODO: should validate the parameters against the strategy.
 
+    def _check_value_type(self, value: _T) -> _T:
```

Comment:
Noted all around. I'm glad you flagged it because I also felt a bit of unease around this - just couldn't quite nail the thought down.

Re: the signature itself, I would take it further and suggest the `@value.setter` signature needs to be tweaked as well
```python
def value(self, value: _T) -> None:
```
to
```python
def value(self, value: Any) -> None:
```

Reply:
Yes indeed.